restapi.go

```diff
@@ -233,26 +235,24 @@ func (self *RestAPI) pushNotification(reqId string, kv map[string]string, perdp
 		case "service":
 			// three keys need to be ignored
 		case "badge":
-			if v != "" {
-				var e error
-				_, e = strconv.Atoi(v)
-				if e == nil {
-					notif.Data["badge"] = v
-				} else {
-					notif.Data["badge"] = "0"
-				}
+			if _, err := strconv.Atoi(v); err == nil {
+				notif.Data["badge"] = v
+			} else {
+				notif.Data["badge"] = "0"
 			}
 		default:
 			notif.Data[k] = v
 		}
 	}
 
+	logger.Debugf("Parameters accepted: %+v", notif)
+
 	if notif.IsEmpty() {
-		logger.Errorf("RequestId=%v From=%v Service=%v EmptyNotification", reqId, remoteAddr, service)
+		logger.Errorf("RequestId=%v From=%v Service=%v NrSubscribers=%v Subscribers=\"%+v\".", reqId, remoteAddr, service, len(subs), subs, err)
```

Comment:
I like how it's explicit about what the problem is versus the fact it's just an error